house/grid.py

```diff
@@ -32,6 +35,28 @@ def get_neighbours(self) -> list:
         
         return nodes
 
+
+    #Check if the GridNode has neighbours to its diagonals in the same plane
+    def has_diagonal_neighbours(self):
+        for dx, dy, dz in [(-1, 0, -1), (-1, 0, 1),(1, 0, 1), (1, 0, -1)]:
+            x = self.x + dx
+            y = self.y + dy
+            z = self.z + dz
+            if (x, y, z) in self.grid.nodes:
+                return True
+        return False
+
+    #Check if the GridNode has diagonal neighbours that are not connected to the same 'building'
+    def has_disconnected_diagonal_neighbours(self):
+        for dx, dy, dz in [(-1, 0, -1), (-1, 0, 1),(1, 0, 1), (1, 0, -1)]:
+            x = self.x + dx
+            y = self.y + dy
```

Comment:
No problem here but do check out vector.py for some useful functions, we can do stuff like point = sum_vectors(coords, dcoords) if you want to skip the work of doing x, y, z manually